### PR TITLE
Add rector [batch]

### PR DESCRIPTION
**Pull request by xepozz**
Added rector/rector to CI and composer.json. Mass creation refers to https://github.com/yiisoft/yii-dev-tool/pull/232

### CODE REVIEW

**Comment by samdark** [comment]
Is this PR ready now?

**Comment by rustamwin** [comment]
Yes.

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
         "source": "https://github.com/yiisoft/auth"
     },
     "require": {
-        "php": "^7.4|^8.0",
+        "php": "^8.0",
```
Comment (vjik) [inline composer.json:20]: Need changelog

File: composer.json [hunk 2]
```diff
@@ -27,6 +27,7 @@
     "require-dev": {
         "nyholm/psr7": "^1.3",
         "phpunit/phpunit": "^9.5",
+        "rector/rector": "^0.14.3",
```
Comment (vjik) [inline composer.json]:
```suggestion
        "rector/rector": "^0.17",
```

File: tests/Method/HttpBasicTest.php [hunk 3]
```diff
@@ -83,9 +81,7 @@ public function testAuthenticationCallbackWithEmptyUsername(): void
     {
         $identityRepository = new FakeIdentityRepository(null);
         $authenticationMethod = (new HttpBasic($identityRepository))
-            ->withAuthenticationCallback(function (?string $username, ?string $password): ?IdentityInterface {
-                return $this->createIdentity(($username ?? 'null') . ':' . ($password ?? 'null'));
-            });
+            ->withAuthenticationCallback(fn (?string $username, ?string $password): ?IdentityInterface => $this->createIdentity(($username ?? 'null') . ':' . ($password ?? 'null')));
```
Comment (vjik) [inline tests/Method/HttpBasicTest.php:84]: Suggest skip rector rule `ClosureToArrowFunctionRector` (see [example](https://github.com/yiisoft/package-template/blob/master/rector.php))

**Review by vjik — CHANGES_REQUESTED** [review]

**Pull request merged**